### PR TITLE
chore: added unit testing for primitive conversions

### PR DESCRIPTION
**Pull request by pollend**
here are some primitive conversion unit test.

### CODE REVIEW

File: tests/conversion.rs [hunk 1]
```diff
@@ -0,0 +1,162 @@
+use rlua::{Lua, ToLua, Value, Result, Table, Error, String};
+
+fn validate_float(verify: Result<Value>, expected: f64) {
+    let verify_unwrap = verify.unwrap();
+    assert_eq!(verify_unwrap.type_name(), "number");
+    match verify_unwrap {
+        Value::Number(value) => assert_eq!(value, expected),
+        _ => assert!(false, "unexpected type"),
```
Comment (jugglerchris) [inline tests/conversion.rs:8]: This could just be `panic!("unexpected type")`.

**Comment by jugglerchris** [comment]
Thanks for the extra tests!
The CI is failing due to formatting - if you could run `cargo fmt` and rerun, that should fix that part.

**Comment by jugglerchris** [comment]
Looks like there may need to be some `#[any(cfg(rlua_lua53), cfg(rlua_lua54))]` to make the new tests pass on Lua 5.1, which doesn't have integers (all values are floats).

**Comment by jugglerchris** [comment]
I'll add the Lua 5.1 support and merge - I've done a few bits like that so shouldn't take too long.

**Comment by jugglerchris** [comment]
I've added my fixes for Lua 5.1 and opened as #232 - I'll give you a chance to check I haven't done anything silly to your changes before merging.  :-)

**Comment by pollend** [comment]
> I've added my fixes for Lua 5.1 and opened as #232 - I'll give you a chance to check I haven't done anything silly to your changes before merging. :-)

sounds good to me.

**Comment by jugglerchris** [comment]
Thanks for doing this!  It's always good to increase test coverage, and this actually found a problem.

**Pull request closed without merging**